### PR TITLE
ensure after handlers are called from method processors

### PR DESCRIPTION
**Pull request by biofractal**
Although I am not entirely sure if it is the way to go, I added calls to `next()` to each of the method processors to ensure the `after` handlers are called. 

see: #2

### CODE REVIEW

File: Resource.js [hunk 1]
```diff
@@ -250,6 +253,7 @@ module.exports = function(app, route, modelName, model) {
             .exec(function(err, items) {
               if (err) return this.respond(res, 500, err);
               res.status(res.statusCode).json(items);
+              next()
```
Comment (travist) [inline Resource.js]: Please add a semi-colon after each ```next()``` statement. While it is valid JavaScript, it is asked for within the code standards that I follow for Node.js projects (http://nodeguide.com/style.html#semicolons).  It also would make jshint complain. Other than that, this looks good to me, and I appreciate your contribution!

Comment (biofractal) [inline Resource.js]: Sorry about the semicolons - bad habits from writing mainly in coffeescript.

I have also pushed up another change for your consideration. For the PUT, POST and DELETE methods I add the affected item to the `res.locals`. This has helped me out with my code as I need the item for socket signalling from the after handlers (see below). The change may be generally useful tho'

```
options.afterPut=(req, res, next)->
	sockets.emit 'PUT', item:res.locals.item
	next()
options.afterPost=(req, res, next)->
	sockets.emit 'POST', item:res.locals.item
	next()
options.afterDelete=(req, res, next)->
	sockets.emit 'DELETE', item: res.locals.item
	next()
```

**Comment by travist** [comment]
I just committed a fix to the "after" handlers getting called.  I basically took what you proposed here and modified it a tad... Please take a look at this commit...

https://github.com/travist/resourcejs/commit/4db96f1dca0d7eac45995428dc524a8bbcd58ca1

This basically does the same thing you were doing here, except it adds a new item to the ```res``` called ```res.resource```.  This has the resource that you can then modify before it gets sent up to the client.

Please let me know if this works for what you need.

One thing we will now need to do is to get your ```patch``` functionality into a separate PR.

Thanks for your efforts.

**Comment by biofractal** [comment]
Thanks for accepting the changes. I am a wee bit rubbish at the git management side of things (but learning) so sorry for messing things up with my half-baked `patch` stuff. This is still a work in progress but hopefully I will have something for your consideration soon.

**Pull request merged**